server/src/main/java/org/elasticsearch/health/components/controller/Controller.java

```diff
@@ -77,4 +80,17 @@ private Controller() {}
         }
         return GetHealthAction.Indicator.of(NO_ELIGIBLE_MASTERS, HealthStatus.RED, "No eligible master nodes");
     }
+
+    static GetHealthAction.Indicator lackOfQuorum(ClusterState clusterState) {
+        DiscoveryNodes nodes = clusterState.nodes();
+        if (nodes.getSize() == 1 && nodes.getNodes().valuesIt().next().isMasterNode()) {
+            return GetHealthAction.Indicator.of(LACK_OF_QUORUM, HealthStatus.GREEN, "Only one master node");
+        }
+        boolean hasQuorum = clusterState.coordinationMetadata()
+            .getLastAcceptedConfiguration()
+            .hasQuorum(clusterState.getNodes().getNodes().values().stream().map(DiscoveryNode::getId).collect(Collectors.toList()));
+        return hasQuorum
+            ? GetHealthAction.Indicator.of(LACK_OF_QUORUM, HealthStatus.GREEN, "Instance can form a quorum")
+            : GetHealthAction.Indicator.of(LACK_OF_QUORUM, HealthStatus.RED, "Not enough nodes with voting rights to form a quorum");
```

Comment:
I have the same question as I did about the other indicator: does this ever return `RED` in production after a node has joined its cluster?

Reply:
It's a good point! Let me test the indicator locally